### PR TITLE
filter CL_VERSION_X_Y macros from reference page see also

### LINKED ISSUE #981: broken refpage links for CL_VERSION_X_Y macros
The refpage links for the `CL_VERSION_X_Y` macros in the "see also" sections all appear to be broken.

Example for [clCreateBuffer](https://registry.khronos.org/OpenCL/sdk/3.0/docs/man/html/clCreateBuffer.html#_see_also) and [CL_VERSION_1_0](https://registry.khronos.org/OpenCL/sdk/3.0/docs/man/html/CL_VERSION_1_0.html) (the first link works, the second doesn't).

I suspect this should redirect to [preprocessorDirectives](https://registry.khronos.org/OpenCL/sdk/3.0/docs/man/html/preprocessorDirectives.html) but I'm not quite sure what magic is needed to make this happen.

@oddhack might be able to help...

### PR DESCRIPTION
**Pull request by bashbaug**
fixes #981 

Instead of trying to make the CL_VERSION_X_Y macros link to the right page, just filter them from the "see also" list, because they aren't adding much value.

I tried to make this somewhat general if we decide to filter other references later, also.  I'm open to suggestions how to do this better though, if desired.

### CODE REVIEW

**Comment by oddhack** [comment]
Did you look at #983 as an alternative?

If doing this I suggest punting the actual exclusion list into a method of clconventions.py

**Comment by bashbaug** [comment]
> If doing this I suggest punting the actual exclusion list into a method of clconventions.py

I moved the exclusion criteria to clconventions.py, if we do end up deciding to do this.

File: scripts/genRef.py [hunk 1]
```diff
@@ -139,6 +139,9 @@ def seeAlsoList(apiName, explicitRefs=None, apiAliases=[]):
                 if dependency is not None:
                     refs.add(dependency)
 
+    # Filter references based on API conventions
+    refs = set([ref for ref in refs if not conventions.filter_ref(ref)])
+
```
Comment (oddhack) [inline scripts/genRef.py:144]: I think you could also, slightly more pythonically, do

> refs = set(filter(lambda ref: conventions.filter_ref(ref), refs))

(which I just learned about on asking "hm, seems like there should be a builtin for this" :-).

**Review by oddhack — COMMENTED** [review]
I don't have any issues with this codewise, but think the issue has been dealt with by the other PR so am slightly against including this.

**Comment by bashbaug** [comment]
I'll convert this to a draft for now.  We can resurrect it after the next release if the linking still seems weird, or we can close it if we're happy with the updated solution.  Thanks!

**Comment by bashbaug** [comment]
I don't think this is needed anymore - closing.

**Pull request closed without merging**